### ISSUE TITLE
Crash on login on 1.21.3-29

### ISSUE THREAD

**Issue by uprial** [PaperSpigot/Paper#11618]
### Stack trace

The server crashes on start
```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000018fdde898, pid=15506, tid=38403
#
# JRE version: OpenJDK Runtime Environment Homebrew (23.0.1) (build 23.0.1)
# Java VM: OpenJDK 64-Bit Server VM Homebrew (23.0.1, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, bsd-aarch64)
# Problematic frame:
# C  [libsystem_platform.dylib+0x6898]  _platform_strcmp$VARIANT$Base+0x8
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/ilyaroslyakov/mine-empty/hs_err_pid15506.log
[28.142s][warning][os] Loading hsdis library failed
#
# If you would like to submit a bug report, please visit:
#   https://github.com/Homebrew/homebrew-core/issues
#


```

### Plugin and Datapack List

no plugins

initial-enabled-packs=vanilla,minecart_improvements,winter_drop

### Paper version

1.21.3-29

### Other

[hs_err_pid15639.log](https://github.com/user-attachments/files/17782201/hs_err_pid15639.log)

**Comment by codebycam** [comment]
This is likely a duplicate of https://github.com/PaperMC/Paper/issues/11579
Can you please try downgrading your Java version to 21?

**Comment by uprial** [comment]
I downgraded the host Java from 23 to 21, and now the Paper server works. Thank you.

**Comment by codebycam** [comment]
> I downgraded the host Java from 23 to 21, and now the Paper server works. Thank you.

No worries! Glad it's sorted.
